modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt

```diff
@@ -247,13 +247,21 @@ interface SequenceKMonadCombine : MonadCombine<ForSequenceK> {
   override fun <A> Kind<ForSequenceK, A>.orElse(b: Kind<ForSequenceK, A>): Kind<ForSequenceK, A> =
     (fix() + b.fix()).k()
 
-  override fun <A : Any> Kind<ForSequenceK, A>.some(): SequenceK<SequenceK<A>> =
+  override fun <A> Kind<ForSequenceK, A>.some(): SequenceK<SequenceK<A>> =
     if (this.fix().isEmpty()) SequenceK.empty()
-    else map { generateSequence { it }.k() }.k()
+    else map { Sequence { object : Iterator<A> {
+      override fun hasNext(): Boolean = true
 
-  override fun <A : Any> Kind<ForSequenceK, A>.many(): SequenceK<SequenceK<A>> =
+      override fun next(): A = it
+    } }.k() }.k()
+
+  override fun <A> Kind<ForSequenceK, A>.many(): SequenceK<SequenceK<A>> =
     if (this.fix().isEmpty()) sequenceOf(emptySequence<A>().k()).k()
-    else map { generateSequence { it }.k() }.k()
+    else map { Sequence { object : Iterator<A> {
+      override fun hasNext(): Boolean = true
```

Comment:
Wait, this passes the tests? lol

I'd expect that you'd need to do some indexed map or something, and use that to check instead.

Reply:
In retrospect, yeah, this makes sense lol